<launch>
	<!--<arg name="environment_1" value="environments/environment_1" />-->

	<rosparam command="load" file="$(find openaggdl)/launch/lion.v.03.yaml" />
	<rosparam command="load" file="$(find openaggdl)/launch/lion_var_types.v.03.yaml" ns="var_types"/>

	<group ns="environments/astrid">
		
		<node 
			pkg="openaggdl" 
			type="topic_filter.py" 
			name="topic_filter_1"/>

		<node 
			pkg="openaggdl" 
			type="sensor_htu21df.py" 
			name="htu21df_temp_1" >
				<param name="rate_hz" value="0.0016666667" type="double"/>
		</node>

		<node 
			pkg="openaggdl" 
			type="sensor_tsl2561.py" 
			name="tsl2561_light_1" >
				<param name="rate_hz" value="0.0016666667" type="double"/>
		</node>

		<node 
			pkg="openaggdl" 
			type="adc_ads1015.py" 
			name="noise" >
				<param name="rate_hz" value="0.0016666667" type="double"/>
				<param name="a0" value="true" type="bool"/>
				<param name="a0_topic" value="noise" type="str"/>
		</node>

		<node 
			pkg="openaggdl" 
			type="adc_ads1015.py" 
			name="air_quality" >
				<param name="rate_hz" value="0.0016666667" type="double"/>
				<param name="a1" value="true" type="bool"/>
				<param name="a1_topic" value="air_quality" type="str"/>
		</node>
		
		<node 
			pkg="openaggdl" 
			type="adc_ads1015.py" 
			name="soil" >
				<param name="rate_hz" value="0.0016666667" type="double"/>		
				<param name="a2" value="true" type="bool"/>
				<param name="a2_topic" value="soil_humidity" type="str"/>

		</node>
		
		<!--	MONGODB	-->

		<node 
			pkg="openaggdl" 
			type="save_db.py" 
			name="sensor_persistence" >
				<!--parametros mongodb-->
				<param name="mongodb_port" value="27017" type="int" />
				<param name="mongodb_host" value="0.0.0.0" type="str" />
				<param name="mongodb_user" value="astrid" type="str"/>
				<param name="mongodb_pwd" value="2018astrid2018" type="str"/>
				<param name="environment_name" value="astrid" type="str"/>
		</node>

		<!--	ACTUADORES	-->

		<node 
			pkg="openaggdl" 
			type="outputs_gpio.py" 
			name="actuators" >
		</node>

		<!--	MQTT	-->

		<node 
			pkg="openaggdl" 
			type="mqtt_connection.py" 
			name="mosquittero" >
				<param name="broker_ip" value="148.202.24.226" type="str" />
		</node>
	
	</group>

</launch>
